clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -78,6 +78,28 @@ public void metricChange(KafkaMetric metric) {
         }
     }
 
+    @Override
+    public void metricRemoval(KafkaMetric metric) {
+        synchronized (LOCK) {
+            KafkaMbean mbean = removeAttribute(metric);
+            if (mbean != null) {
+                if (mbean.metrics.isEmpty())
+                    unregister(mbean);
+                else
+                    reregister(mbean);
```

Comment:
Can you explain the reason behind reregistering? On metric removal, you should simply unregister the Mbean right?